README.md

```diff
@@ -209,27 +209,64 @@ Authorization in `root:redhat-appstudio` and `root:redhat-hacbs` workspaces in C
 
 For access to the OpenShift staging cluster, the user must be added to the `stage` team in the `redhat-appstudio-sre` Github organization.
 
-## Enabling Monitoring for workload cluster
+## Monitoring for workload clusters
 
-Prometheus Operator is deployed through ArgoCD. This is configure to run in `appstudio-workload-monitoring` namespace. 
-This prometheus is accessed through exported route, which is secured by github oauth. Github oauth secrets are configured in the cluster through secret
+Note:
 
-`prometheus-proxy-config`
+This section refers to "Grafana cluster" and "Prometheus cluster" as the clusters on which Grafana and Prometheus are deployed, respectively. In a multi-cluster topology, there will be a single cluster on which Grafana is deployed, whereas Prometheus will be deployed on all clusters where collecting metrics is needed.
+
+### Setup
+
+First, create the `appstudio-workload-monitoring` namespace if it does not exist yet:
+
+```
+$ oc create namespace appstudio-workload-monitoring
+```
+
+#### OAuth2 proxy secrets
+
+Both Prometheus and Grafana UIs are protected by an OAuth2 proxy running as a sidecar container and which delegates the authentication to GitHub. 
+Users must belong to the [Red Hat Appstudio SRE organization](https://github.com/redhat-appstudio-sre) team configured in the OAuth2 proxy to be allowed to access the Web UIs.
+
+Create the secrets with the following commands:
+
+```
+$ ./hack/setup-monitoring.sh oauth2-secret prometheus-oauth2-proxy $PROMETHEUS_GITHUB_CLIENT_ID $PROMETHEUS_GITHUB_CLIENT_SECRET $PROMETHEUS_GITHUB_COOKIE_SECRET
+
+$ ./hack/setup-monitoring.sh oauth2-secret grafana-oauth2-proxy $GRAFANA_GITHUB_CLIENT_ID $GRAFANA_GITHUB_CLIENT_SECRET $GRAFANA_GITHUB_COOKIE_SECRET
+```
```

Comment:
I think you should specify here that the promethus-oauth2-proxy secret needs to be created for each Prometheus cluster.  And do you need a different "OAuth application" for each cluster or is it the same one for all clusters?

And then specify that you only need to create the grafana-oauth2-proxy secret once, on the Grafana cluster.

Reply:
good point, let me clarify that! And yes, it's a different OAuth Application for each cluster, indeed

Reply:
done in 13337c0

Reply:
this means we need to update the env variables before running for each prometheus cluster. could we put this somewhere?

Reply:
> could we put this somewhere?

did you mean documenting it? If so, please see in 98c9d0f

Reply:
These changes look good to me, thanks.